rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryInstances.scala

```diff
@@ -203,22 +203,115 @@ object HistoryInstances {
       FlatMap[F].tailRecM((fullPath, trieNodesOnPath, none[Trie]))(go)
     }
 
-    def process(actions: List[HistoryAction]): F[History[F]] =
-      // TODO this is an intermediate step to reproduce all the trie behavior
-      // will evolve to a fold based implementation with partial tries
+    def findUncommonNode(
+        currentPath: KeyPath,
+        previousPath: KeyPath,
+        previousRoot: Trie
+    ): F[Option[Blake2b256Hash]] = {
+      def commonPrefixWithLastDiverging(path: KeyPath, affix: ByteVector): Boolean =
+        affix.toSeq.startsWith(path.dropRight(1))
+
+      def handlePointerToPointerBlock(
+          ptr: ValuePointer
+      ): F[Either[(KeyPath, Trie), Option[Blake2b256Hash]]] =
+        ptr match {
```

Comment:
I can't see `Left` being ever returned, so it would make sense to get rid of the `Either` from the return type. Also enclosing over `F` is superfluous as this looks like a pure function. Is there a better name than `handle`? Maybe `follow`?